torch/csrc/jit/tensorexpr/codegen.h

```diff
@@ -111,6 +119,7 @@ class TORCH_API CodeGen {
   std::vector<BufferArg> buffer_args_;
   at::Device device_ = at::kCPU;
   std::string kernel_func_name_ = "func";
+  std::vector<BufPtr> buffers_to_alloc_;
```

Comment:
The names are getting quite confusing now. If one reads this code after this PR lands, they would see `buffer_to_alloc`, `buf_allocs`, `interm_bufs`, `buf_args` and many of those things, despite appearing similar, mean very different things. Let's rename them to better reflect what they hold. E.g. one of them can be `preallocated_bufs` - i.e. a set of list of buffers that don't have Allocate/Free statements and need to be preallocated outside of the codegen (i.e. in the Kernel). The other one can be `dynamically_allocated_bufs` - those are the intermediate bufs that have Allocate/Free/Alias stmt for them. You could choose different names if you'd like but please try to assess the code from a position of a person reading it for the first time.